### PR TITLE
[SBS-2][SBS-55] feat: integrate api add product to cart

### PR DESCRIPTION
**Pull request by phdhuy**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by phdhuy** [comment]
https://github.com/superbadteam/superbad-store_mobile/assets/95920128/2a10c8b8-3a1a-477a-841f-89d52c8da6ea

File: app/screens/ProductDetailScreen/ProductDetailScreen.tsx [hunk 1]
```diff
@@ -31,13 +31,15 @@ const ProductDetailScreen = () => {
...
+  const { addProductToCart, isMutating } = useAddProductToCart();
+
 
-  const navigation = useNavigation<any>();
   const handleAddToCart = () => {
-    navigation.navigate("CartScreen");
+    addProductToCart();
   };
 
+
```
Comment (phamhongphuc1403) [inline app/screens/ProductDetailScreen/ProductDetailScreen.tsx]: dùng trực tiếp thay vì bỏ vào 1 func mới

Comment (phdhuy) [reply app/screens/ProductDetailScreen/ProductDetailScreen.tsx]: done

File: app/screens/ProductDetailScreen/ProductDetailScreen.tsx [hunk 2]
```diff
@@ -106,7 +102,7 @@ const ProductDetailScreen = () => {
       {/* Buttons */}
       <View style={$buttonContainer}>
         <TouchableOpacity style={[$button, $addToCartButton]}
-        onPress = {handleAddToCart}>
+          onPress={() => addProductToCart()} disabled={isMutating}>
```
Comment (phamhongphuc1403) [inline app/screens/ProductDetailScreen/ProductDetailScreen.tsx:105]: ?

**Review by phamhongphuc1403 — APPROVED** [review]

**Review by huynamboz — APPROVED** [review]
OK

**Pull request merged**